src/components/OSCALLoader.js

```diff
@@ -197,17 +183,28 @@ export default function OSCALLoader(props) {
   // Note: the empty deps array [] means
   // this useEffect will run once
   // similar to componentDidMount()
-  useEffect(() => {
-    if (!isRestMode) {
-      loadOscalData(oscalUrl);
-    }
-    return () => {
+  useEffect(
+    () => () => {
       unmounted.current = true;
-    };
-  }, []);
+    },
+    []
+  );
+
+  useLayoutEffect(() => {
+    if (isRestMode) {
+      setOscalUrl(null);
+      setIsLoaded(true);
+      setIsResolutionComplete(true);
+    } else {
+      setOscalUrl(props.oscalObjectType.defaultUrl);
+      loadOscalData(props.oscalObjectType.defaultUrl);
+      setIsLoaded(false);
+      setIsResolutionComplete(false);
+    }
+  }, [isRestMode]);
 
   let form;
-  if (props.renderForm) {
+  if (props.renderForm && oscalUrl) {
```

Comment:
This feels like an odd thing to require in the condition. Even if (for some reason) `oscalUrl` were null/undefined (or otherwise falsey), we probably want the form to still be created.

Imagine a user presses enter on the form while it's empty! Would we just remove the form for that?